components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/ha/HAManager.java

```diff
@@ -193,16 +194,47 @@ void changeToActive() {
             }
         }
         if (!liveSyncEnabled && !isActiveNodeOutputSyncManagerStarted) {
-            ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
-            scheduledExecutorService.scheduleAtFixedRate(new ActiveNodeOutputSyncManager(
-                            sinkHandlerManager, recordTableHandlerManager, clusterCoordinator), 0, outputSyncInterval,
-                    TimeUnit.MILLISECONDS);
+            activeNodeScheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+            activeNodeScheduledExecutorService.scheduleAtFixedRate(new ActiveNodeOutputSyncManager(
+                            sinkHandlerManager, recordTableHandlerManager, clusterCoordinator),
+                    0, outputSyncInterval, TimeUnit.MILLISECONDS);
             isActiveNodeOutputSyncManagerStarted = true;
         }
         NodeInfo nodeInfo = StreamProcessorDataHolder.getNodeInfo();
         nodeInfo.setActiveNode(isActiveNode);
     }
 
+    /**
+     * Starts Publisher Syncing of Passive Node
+     * Starts scheduled state sync of Passive Node from Active Node
+     * Updates Coordination Properties with Advertised Host and Port
+     */
+    void changeToPassive() {
```

Comment:
When we are changing to passive , don't we need to stop any schedulers started in active mode ?

Reply:
We already shutdown the active node's schedulers in line 220

Reply:
Ack